src/librustc_driver/driver.rs

```diff
@@ -804,6 +799,10 @@ pub fn phase_2_configure_and_expand<F>(sess: &Session,
         })
     })?;
 
+    time(time_passes,
+         "early lint checks",
+         || lint::check_ast_crate(sess, &krate));
```

Comment:
So, with this PR lints can no longer be reported *during* lowering to HIR, right?
It's too late for "early lints" and too early for "late lints", but previously they could be reported as "late".
What if I have a lint that should work this way?